src/main/java/eventprocessing/responseservices/ResponseService.java

```diff
@@ -0,0 +1,29 @@
+package eventprocessing.responseservices;
+import com.google.gson.Gson;
+import com.google.gson.JsonSyntaxException;
+import eventprocessing.models.Message;
+import eventprocessing.models.Response;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+
+public class ResponseService {
+
+    private final Logger logger = LogManager.getLogger("ResponseService");
+
+    public Response parseResponse(String jsonString) {
+        Gson gson = new Gson();
+        Response responseObject = null;
+        try {
+            responseObject = gson.fromJson(jsonString, Response.class);
+            String messageString = responseObject.getMessageString();
+            Message messageObject = gson.fromJson(messageString, Message.class);
+            responseObject.setMessage(messageObject);
+        } catch (IllegalStateException | JsonSyntaxException e) {
+            logger.warn("Invalid JSON string received" + e.getMessage());
```

Comment:
Logging tip: all the log4j logging methods (`error`, `warn`, etc.) have an overload which takes a `Throwable` as well as a message, for example: https://logging.apache.org/log4j/2.0/log4j-api/apidocs/org/apache/logging/log4j/Logger.html#error(org.apache.logging.log4j.message.Message,%20java.lang.Throwable)

This gives you better logs, because it includes the stack trace as well as the message.